clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -156,7 +156,7 @@ public static int utf8Length(CharSequence s) {
      * Read the given byte buffer into a byte array
      */
     public static byte[] toArray(ByteBuffer buffer) {
-        return toArray(buffer, 0, buffer.limit());
```

Comment:
It seems that we should either use (1) 0 and buffer.limit() or (2) buffer.position() and buffer.remaining(), but not mixing the two? For (2), typically the buffer is in the write mode. In the read mode, (1) is probably what we want?